### PR TITLE
Setup master as a node too

### PR DESCRIPTION
**Pull request by mbforbes**
Addresses #8315 (I hope).

I'm running GCE and GKE e2es for another PR currently, so it will be at least 45 minutes before I can run them on this PR to confirm it fixes the issue. I'll add a comment on this PR once I run them.

### CODE REVIEW

**Comment by mbforbes** [comment]
+cc @fgrzadkowski 
+cc @roberthbailey 

Happy to accept a review.

**Comment by vmarmol** [comment]
Niiiice :)

/cc @dchen1107 @vishh

**Comment by roberthbailey** [comment]
LGTM.

**Comment by roberthbailey** [comment]
Will merge once e2e is passing.

**Comment by wojtek-t** [comment]
Thanks @mbforbes

**Comment by vishh** [comment]
Thanks for this PR @mbforbes

File: cluster/gce/configure-vm.sh [hunk 1]
```diff
@@ -559,11 +559,8 @@ if [[ -z "${is_push}" ]]; then
   set-kube-env
   [[ "${KUBERNETES_MASTER}" == "true" ]] && mount-master-pd
   create-salt-pillar
-  if [[ "${KUBERNETES_MASTER}" == "true" ]]; then
-    create-salt-master-auth
```
Comment (fgrzadkowski) [inline cluster/gce/configure-vm.sh]: The error I've seen was "certificate is valid for 104.155.29.206, not 10.240.81.1". AFAIU it means the apiserver was using a certificate created for 104* IP, wheres kubelet was using 10* IP to connect to it. AFAIU it's not enough to call ```create-salt-node-auth```, but you also have to set apiserver address by calling ```salt-set-apiserver```. Am I missing something?

Comment (roberthbailey) [inline cluster/gce/configure-vm.sh]: I think I can see in the salt why this is happening. @zmerlynn 

configure-vm.sh sets the apiserver grain on the minions but not on the master (https://github.com/GoogleCloudPlatform/kubernetes/blob/master/cluster/gce/configure-vm.sh#L541). So when salt is determining what address to pass the kubelet for the master, it hits the third or fourth case in `cluster/saltbase/salt/kubelet/default` which ends up picking the 10. address (https://github.com/GoogleCloudPlatform/kubernetes/blob/master/cluster/saltbase/salt/kubelet/default#L11-L12 or https://github.com/GoogleCloudPlatform/kubernetes/blob/master/cluster/saltbase/salt/kubelet/default#L14-L15). I can't tell which case is happening, but it seems likely that one (or both) of them would result in the internal address being used.

That means that this change won't help because it isn't setting the apiserver grain on the master VM. 

In addition, in #6949 I'm explicitly detaching the kubelet on the master from the cluster, so this problem will go away. When we re-add the option to attach it, we should plumb the appropriate configuration bits so that the master looks just like any other node in the cluster.

**Comment by roberthbailey** [comment]
@mbforbes -- I'm going to close this because I don't think it will help and it should become obsolete once my PR is merged (hopefully in the next day or so). Please re-open if you'd still like to see this get merged.

File: cluster/gce/configure-vm.sh (hunk 1 above)
Comment (zmerlynn) [inline cluster/gce/configure-vm.sh]: Yeah, that was a change by @ArtfulCoder to explicitly use the 10.* IP of the master (the external ipv4 address). The master can't actually talk to itself on its own 104.* address without some other network shenanigans, so a 104.* address, on the master, just won't work, unfortunately.

Comment (mbforbes) [inline cluster/gce/configure-vm.sh]: > the 10.* IP of the master (the external ipv4 address)

do you mean s/external/internal there?

@roberthbailey thanks for looking into this. I didn't see how this would fix the internal vs external IP address issue, but I figured I'd prepare this essentially as a revert because my previous PR was mentioned as a possible root cause.

Would it be useful to add the `salt-set-apiserver` call to this and get it in before your #6949 lands? Any idea how long it will be? This absolutely floods the master's kubelet logs making them pretty hard to parse.

Comment (roberthbailey) [inline cluster/gce/configure-vm.sh]: > Would it be useful to add the salt-set-apiserver call to this

That will work as long as $MASTER_NAME isn't set to the external IP of the master (as @zmerlynn mentioned). 

I'm hoping that @erictune will have time today to look at #6949 now that it passes e2e tests.

Comment (zmerlynn) [inline cluster/gce/configure-vm.sh]: | do you mean s/external/internal there?

Yes, sorry.
